### ISSUE TITLE
State syncing always failure

### ISSUE THREAD

**Issue by garyyu** [mimblewimble/grin#2246]
Since the chain get to `10,132` height, the state sync starts to work for new node.
But I see it always fail and download again and again:
```
20181228 03:00:28.239 DEBUG grin_chain::txhashset::txhashset - txhashset: verified 5108 rangeproofs, pmmr size 30741, took 8s
20181228 03:00:28.239 DEBUG grin_chain::txhashset::txhashset - Error returned, discarding txhashset extension: Transaction Error
 Cause: Unknown
 Backtrace:
20181228 03:00:28.241 ERROR grin_servers::common::adapters - Failed to save txhashset archive: Transaction Error
 Cause: Unknown
 Backtrace:
20181228 03:00:28.241 DEBUG grin_p2p::peers - Received a bad txhashset data from 185.8.164.22:13414, the peer will be banned
20181228 03:00:28.244 ERROR grin_servers::grin::sync::state_sync - state_sync: error = Chain(Error { inner:

Transaction Error }). restart fast sync
20181228 03:00:28.254 DEBUG grin_p2p::peers - Banning peer 185.8.164.22:13414
20181228 03:00:28.254 DEBUG grin_p2p::peer - Sent ban reason BadTxHashSet to 185.8.164.22:13414
20181228 03:00:28.254 DEBUG grin_p2p::protocol - handle_payload: txhashset archive for 015549aa at 7242, DONE. Data Ok: false
```

**Comment by ignopeverell** [comment]
Looks like this is the culprit:

https://github.com/mimblewimble/grin/blob/4f0582a9843b55e55a15a087fe7d319d94786d01/core/src/core/transaction.rs#L1369

The genesis for floonet uses what's in the `else` branch instead of the `if` branch for anything floonet. That code was in flux around the floonet release so I'm guessing we changed it right after release or something to that effect. Should be fixed with the floonet relaunch.

**Comment by ignopeverell** [comment]
Closing this as we relaunched floonet.

**Comment by garyyu** [comment]
I reopen it too quick, and it proves that the new state sync problem is not related to this old one. It should not be re-opened.

The new state sync problem is a regression imported by #2269 , and should be fixed by #2289 .

So let me close this issue again.
Sorry for the noise~
